Sonal/more templates work (#4658)
forcedotcom/salesforcedx-vscode: docs/_articles/en/apex/writing.md

- [Apex Code Snippets](https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-apex/snippets/apex.json)
- [Code Snippets for LWC development](https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-lwc/snippets/lwc.json) 

### Example Custom Snippet

> The real power of snippets lies in being able to customize snippets for your own use. Follow these steps to create a simple custom Apex snippet that lets you quickly write a simple SOQL query:

Reviewer: nit: we're using simple twice

> 1. Run the `SFDX: Configure User Snippets` command from the Command Palette. 

Reviewer 2: @sbudhirajadoc - I'd have to take a look but don't we use bold for menu command to mirror uicontrol? Or did you change the style?

Author: Thanks for catching that. Those commands need to be in bold. I'll make the udpate. Thanks!

2. Select ``apex.json`` to open the file.
3. Add this snippet code to the file:
   
   ```
   "SOQL" : {
        "prefix": "soql",
        "body": [
            "[SELECT ${1:field1, field2} FROM ${2:SobjectName} WHERE ${3:clause}];"
        ],
        "description": "Apex SOQL query"
    }
   ```
4. Save the file.

> 5.  Use this snippet in your Apex class file by simply typing "soql" and selecting to add this snippet to your code.

Reviewer 2: ```suggestion
5.  Use this snippet in your Apex class file by typing "soql" and selecting to add this snippet to your code.
```

> 6.  The code snippet `` [SELECT field1, field2 FROM SobjectName WHERE clause];`` is added to your apex file. 

Reviewer 2: ```suggestion
6.  The code snippet `` [SELECT field1, field2 FROM SobjectName WHERE clause];`` is added to your Apex file. 
```

- [Apex Code Snippets](https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-apex/snippets/apex.json)
- [Code Snippets for LWC development](https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-lwc/snippets/lwc.json) 

> If you want to define your own code snippets, read [Snippets in Visual Studio Code](https://code.visualstudio.com/docs/editor/userdefinedsnippets).

Reviewer 3: Nice! I like adding more specifics on custom snippets here, rather making users wander off to find this information.

### Example Custom Snippet
The real power of snippets lies in being able to customize snippets for your own use. Follow these steps to create a custom Apex snippet that lets you quickly write a simple SOQL query:

1. Run the `SFDX: Configure User Snippets` command from the Command Palette. 
2. Select ``apex.json`` to open the file.
3. Add this snippet code to the file:
   
   ```
   "SOQL" : {
        "prefix": "soql",
        "body": [
            "[SELECT ${1:field1, field2} FROM ${2:SobjectName} WHERE ${3:clause}];"
        ],
        "description": "Apex SOQL query"
    }
   ```
4. Save the file.
5.  Use this snippet in your Apex class file by simply typing "soql" and selecting to add this snippet to your code.
6.  The code snippet is added to your apex file: 

   `` [SELECT field1, field2 FROM SobjectName WHERE clause];``
  

See [Snippets in Visual Studio Code](https://code.visualstudio.com/docs/editor/userdefinedsnippets) for more information about snippets.

## Custom Metadata Templates

> In addition to the generic snippets feature available in VS Code, you can use custom templates to create new metadata for an Apex class or trigger. See [Custom Code Templates](./en/user-guide/byotemplate) for information on how to set up your custom templates. Here's an example of an update to an Apex class:

Reviewer 3: ```suggestion
In addition to the generic snippets feature available in VS Code, you can use custom templates to create new metadata for an Apex class or trigger. See [Custom Code Templates](./en/user-guide/byotemplate) for information on how to set up your custom templates. Here's an example of using a template for your Apex classes:
```

1. Edit the `DefaultApexClass.cls` file with your custom code:

```

> //Copyright (c) <year><copyright holder>

Reviewer 3: It looks like this template doesn't match the generated code below
